trajectory_executor/com_serial.py

```diff
@@ -175,15 +174,30 @@ def sensor_data_request_callback(self, request: SensorDataRequest, response: Sen
             response.at_goal = data_packet['at_goal']
             break
           else:
-            # bad packet, retry request
-            return self.sensor_data_request_callback(request, response)
+            # bad packet, set values to -1 and let executor handle it
+            # TODO: We should probably add a "error" value in the message for
+            #       the executor to use
+            response.roll = -1
+            response.pitch = -1
+            response.yaw = -1
+            response.l_hip = -1
+            response.l_knee = -1
+            response.r_hip = -1
+            response.r_knee = -1
+            response.l_shoulder = -1
+            response.l_elbow = -1
+            response.r_shoulder = -1
+            response.r_elbow = -1
+            response.at_goal = -1
+            break
 
     return response
 
   def send_joint_angles_callback(self, msg: JointAngles):
     """Send new joint angles to Arduino for execution"""
     # Split joint angles into two numbers since max value of a byte is 255
     ja_bytes = []
+    ja_bytes.append(0) # data request byte
```

Comment:
Not 100% sure I'm reading this correctly, but this byte is basically just a flag telling if it's a response or a request, correct?

Reply:
Correct. 0: sending joint angles. 1+: requesting sensor data

Reply:
Let's declare a variable that holds this value?